src/main/g8/src/main/scala/$package$/DemoParallelism.scala

```diff
@@ -1,48 +1,75 @@
+   * @param parallelism Concurrent processes
+   */
+  def apply(iterations: Long, parallelism: Int): Task[Double] = for {
+    // log that the work is starting
+    _ <- Task {
+      logger.info(f"Will approximate Pi with \$iterations%,d iterations of random bubble-filling and up to \$parallelism%,d parallel processes...")
+    }
 
-    // Scheduler for parallelism:
-    implicit val s: Scheduler = Scheduler.forkJoin(parallelism, maxThreads = 1000)
+    // create scheduler resource based on desired parallelism (it will be closed once we have finished our calculations)
+    schedulerResource = Resource.make(acquire = Task {
+      Scheduler.forkJoin(parallelism = parallelism, maxThreads = 1000)
+    })(release = s => Task { s.shutdown() })
 
-    // Approximation of pi using iteration:
-    val pi = approximatePi(iterations).runToFuture
+    // then use the scheduler to...
+    calculate <- schedulerResource.use(scheduler => {
+      // calculate an approximation of pi
+      approximatePi(parallelism, iterations)
+        // measure how long it took to perform the calculation
+        .timed
+        // set a timeout of 30 seconds
+        .timeout(30.seconds)
+        // insure the process runs on the provided scheduler
+        .executeOn(scheduler)
+    })
 
-    logger.info(s"Pi is roughly \${Await.result(pi, atMost = 30.seconds)}")
+    // log the results
+    timeTook = calculate._1
+    pi = calculate._2
+    _ <- Task {
+      logger.info(s"Pi is roughly \${pi}. (Took \${timeTook.toMillis}ms)")
+    }
+  } yield {
+    pi
   }
 
   /**
     * Approximate pi via [[https://en.wikipedia.org/wiki/Approximations_of_%CF%80#Summing_a_circle's_area Monte Carlo method]].
     */
-  def approximatePi(iterations: Long, seed: Int = 13): Task[Double] = {
+  def approximatePi(parallelism: Int, iterations: Long, seed: Int = 13): Task[Double] = {
     val random = new Random(seed)
+
     // Kernel: projects a random point to detect if within bounds of a circle:
     val detectCircularity: Task[Long] = Task { (random.nextDouble(), random.nextDouble()) }
       .map { case (x, y) => (x * 2 - 1, y * 2 - 1) }
       .map { case (x, y) => if (x * x + y * y <= 1) 1 else 0 }
 
-    // The list of all tasks needed for execution:
-    val tasks = for { _ <- 0L until iterations } yield detectCircularity
-    // Gathering tasks, then summing the final result:
-    val aggregate = Task.parSequence(tasks).map(_.sum)
-    // Calculating pi from the aggregate:
-    aggregate.map { 4.0 * _ / (iterations - 1) }
+    // create a stream of "observable events" matching the desired iterations
+    Observable.range(0, iterations)
+      // for each iteration, create a projection and check it
+      .mapParallelUnordered(parallelism)(_ => detectCircularity)
+      // sum the results of each detection
+      .sumL
+      // finally, calculate pi from the aggregate sum
+      .map(sum => {
+        4.0 * sum / (iterations - 1)
+      })
```

Comment:
So here's some fun: We can actually still make this code several orders of magnitude more performant. All we would have to do is batch iterations into a single Task. Like this:

```
    // create a stream of "observable events" matching the desired iterations
    Observable.range(0, iterations)
      // batch iterations into groups of 100 thousand or less
      .bufferTumbling(100000)
      // for each batch of iterations, create a projections for each one and check it
      .mapParallelUnordered(parallelism)(batchIterations => {
        val batchedComputations = batchIterations.map(_ => detectCircularity)
        Task.sequence(batchedComputations).map(_.sum)
      })
      // collect all the results, then sum the results of each event
      .sumL
      // finally, calculate pi from the aggregate sum
      .map(sum => {
        4.0 * sum / (iterations - 1)
      })
```

*Q*: What kind of an effect does this have? 
*A*: Well, for starters, we can start running approximations with iterations of 10,000,000 in less than a second which is pretty legit. That's over 20x more performant than even the best case scenario of the NEW code, let alone the old.

*Q*: So why don't we do it?
*A*: If the goal of this project was to make an insanely performant, accurate, and long-running approximation of pi, we probably wouldn't be using graal native, since the JVM is gonna blow the doors off graal native in any application with a long runtime. I think the implementation of this PR does a couple things really well.
Firstly, it illustrates a clean way of performing parallel processing that works very well in graal native. Most importantly though, it has very reasonable and predictable performance based on the inputs you provide it. 

*TL;DR* If your goal is to show people a "real-world" example how parallelism in graal can be implemented and how well it performs, I think you'd have a hard time coming up with a better example than what we have here.

Reply:
> Q: What kind of an effect does this have?
> A: Well, for starters, we can start running approximations with iterations of 10,000,000 in less than a second which is pretty legit. That's over 20x more performant than even the best case scenario of the NEW code, let alone the old.

This is awesome.  To be honest, this makes me want to start a benchmarking competition.  Maybe we can have a "pi eating race" in a future version of this repo that includes a JMH harness in SBT?  JMH is actually useful, _and we can have some fun while we're at it, right?_

> Q: So why don't we do it?
> A: If the goal of this project was to make an insanely performant, accurate, and long-running approximation of pi, we probably wouldn't be using graal native, since the JVM is gonna blow the doors off graal native in any application with a long runtime. I think the implementation of this PR does a couple things really well.
> Firstly, it illustrates a clean way of performing parallel processing that works very well in graal native. Most importantly though, it has very reasonable and predictable performance based on the inputs you provide it.

Agreed.  Your "old new" implementation is already great, with the advantage of being very readable.  This is important, since the Monte Carlo method of Pi approximation is already a bit "abstract".  But if I were to highlight places for improvement in the "new new" implementation in your comment...
1. An observable range and a [tumblingBuffer](https://monix.io/api/current/monix/reactive/Observable.html#bufferTumbling(count:Int):monix.reactive.Observable[Seq[A]]) are ordered and deterministic - and both of those things have a cost in terms of memory allocation.  If the number of iterations on each process is fixed, we would need no buffering and no observing fundamentally - we are only using those Monix constructs out of convenience. This goes all the way down to the random calculation encapsulating a tuple - you technically don't even need that data structure.  I also think that for the purposes of pure optimization, more batches isn't necessarily better - eliminating arbitrary boundaries and delegating the allocation of resources to the scheduler is better.
2. The parallelism in `mapParallelUnordered(parallelism)` is also used to configure the scheduler this runs in, effectively coupling concurrency and parallelism.  Depending on if this is running on the JDK or on Graal _(or on Javascript)_ and how the "Threads" are handled _([for instance](https://monix.io/docs/current/execution/scheduler.html#execution-model), a single process that delegates to setTimeout)_, this may have some adverse consequences to speed.

Reply:
:+1: on the idea of a benchmark competition. That sounds like a good bit of fun! I actually only recently started messing with JMH and I gotta say, I totally agree that it can be a ton of fun AND hella useful

Reply:
> I were to highlight places for improvement in the "new new" implementation in your comment...

Some 5-head points there :+1: and even more reason to go about the benchmarking contest. Based on your comments, I'm already seeing yet ANOTHER set of ways that performance could be improved... particularly for graal-native. If there is gonna be a contest, I'm keeping my cards close to my chest :joy: